src/parse-rules/default/markdown-to-wikitext.js

```diff
@@ -51,54 +51,29 @@ function parseBlockquoteSecondPass(line, index, arr) {
   return newLine
 }
 
-function parseLink(line) {
+function link(line) {
   return line.replace(/\[([^\[]*)\]\(([^\(]*)\)/g, `[$2 $1]`); /* [textl](url) */
 }
 
-function parseBold(line) {
+function bold(line) {
   return line.replace(/[*]{2}/g, `'''`)
 }
 
-function parseItalic(line) {
+function italic(line) {
   return line
     .replace(/([^\\]{0,1})_/g, `$1''`) /* first take `_` in markdown meanings and translate it to '' and '' */
     .replace(/\\[_]/g, '_') /* then take all simple symbol `_` who is screened and write it without `\` */
 }
 
-function skip(line) {
-  return line
-}
-
-/*
-options = {
-  skipOrderedList: true,
-  skipHeadings: true,
-  skipUnorderedList: true,
-  skipCode: true,
-  skipBlockquote: true,
-  skipLink: true,
-  skipBoldAndItalic: true,
-
-  customRules: []
-}
-*/
-module.exports = (text, options = {}) => {
-  let newText = text.split('\n')
-
-  if (options.customRules) {
-    options.customRules.forEach(rule => newText = newText.map(rule))
-  }
-
-  return newText
-    .map(options.skipHeadings && skip || parseHeadings)
-    .map(options.skipUnorderedList && skip || parseUnorderedList)
-    .map(options.skipOrderedList && skip || parseOrderedList)
-    .map(options.skipCode && skip || parseCode)
-    .map(options.skipBlockquote && skip || parseBlockquoteFirstPass)
-    .map(options.skipBlockquote && skip || parseBlockquoteSecondPass)
-    .map(options.skipLink && skip || parseLink)
-    .map(options.skipBoldAndItalic && skip || parseBoldAndItalic)
-    .map(options.skipBoldAndItalic && skip || parseBold)
-    .map(options.skipBoldAndItalic && skip || parseItalic)
-    .join('\n')
-}
+module.exports = [
```

Comment:
I would prefer to have an object with methods here:

```
markdownToWikitextRules = {
  ['headings']: (line) => { 
    ... 
  },

  ...

  ['italic']: (line) => { 
    ... 
  }
}

module.exports = markdownToWikitextRules;
```

So rules tokens are explicit now and could be extracted into constant enumeration and reused:

```
RULES.ITALIC = 'italic';
...
  [RULES.ITALIC]: (line) => { 
    ... 
  }
```